src/types/markers.js

```diff
@@ -48,42 +48,145 @@ export type PaintProfilerMarkerTracing = ProfilerMarkerTracing & {
     | 'Composite',
 };
 
-export type GCMinorMarkerPayload = {
-  type: 'GCMinor',
-  startTime: Milliseconds,
-  endTime: Milliseconds,
-  // nursery is only present in newer profile format.
-  nursery?: {|
-    reason?: string,
-    status?: string,
-  |},
+export type PhaseTime = { [phase: string]: Milliseconds };
+
+export type GCSliceData = {
+  // Slice number within the GCMajor collection.
+  slice: number,
+
+  pause: Milliseconds,
+  when: Milliseconds,
+
+  // The reason for this slice.
+  reason: string,
+
+  // The GC state at the start and end of this slice.
+  initial_state: string,
+  final_state: string,
+
+  // The incremental GC budget for this slice (see pause above).
+  budget: Milliseconds,
+
+  // The number of the GCMajor that this slice belongs to.
+  major_gc_number: number,
+
```

Comment:
nit: please add a comment explaining the reason `trigger_amount` and `trigger_threshold` are optional